### DISCUSSION TITLE
How to show error message underneath text input with useFieldArray

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5386] [Q&A] [answered]
If this was a normal field I would display the error like:

`{errors.lineItem?.message}`

But I am using useFieldArray and the fields are in a loop and it doesn't work. I probably need to use the index but not sure how?

```
const schema = object().shape({
  items: array()
    .of(
      object().shape({
        lineItem: string().required("Some text is required"),
        qty: number().required("Qty required"),
        price: number().required("Price required"),
      })
    )
    .required(),
});

{fields.map((field, index) => (
      <Form.Group controlId="lineItem">
        <Form.Control
            type="text"
              defaultValue={field.lineItem}
               placeholder="Line Item"
               {...register(`items.${index}.lineItem`)}
                 />
                {errors.lineItem?.message}
     </Form.Group>
))}
```

I also tried this:

`errors.items[index]`

But I get error: Cannot read property '0' of undefined

**Comment by bluebill1049** [comment, marked as answer]
try that `errors.items?[index]`

**Reply by ghost** [reply]
That gave me an error. I did manage to get it working with this though. However, it seems a bit verbose. Any way to trim it down? Not a big deal if not.

`{errors?.items?.[index]?.lineItem?.message}`

**Reply by wdfinch** [reply]
{errors.items.[index].lineItem.message} is how you would get the error for an individual item in the field array. 

There's no way to trim it because nothing else makes sense. Your asking for the error message for a specific field in a field array. Therefore the above references exactly what you are asking for.

The only other options to trim it are:

1. use `Controller` rather than `register`. `Controller` render method will provide the error in a more concise way. 
2. Save a reference in your component to part of the error object. EX. `const fieldArrayErrors = errors.items.[index]`. This isn't more concise, but just moves things around.

Lastly, if you want array errors from a schema, for example, you want to validate that your array has one element, you can't get the error message unless you write a custom resolver. The validation will work still. You can vote for this behavior here #4356
